src/main/scala/mesosphere/marathon/core/appinfo/GroupInfoService.scala

```diff
@@ -0,0 +1,17 @@
+package mesosphere.marathon.core.appinfo
+
+import mesosphere.marathon.state.Group
+
+import scala.concurrent.Future
+
+/**
+  * Queries for extended group information.
+  */
+trait GroupInfoService {
+
+  /**
+    * Query info for an existing group.
+    */
+  def queryForGroup(group: Group, appEmbed: Set[AppInfo.Embed], groupEmbed: Set[GroupInfo.Embed]): Future[GroupInfo]
```

Comment:
I am not sure if I agree. You call this "queryForGroup", similar to the other methods. Yet, the other methods take care of loading all required data. This is more like "enrichGroupInfo" or something like this

The selector-based approach would at least use the same principle for groups that is used for apps. It feels strange to use something different for such a similar method in the same implementation class.